### PR TITLE
Check data APIs nullability for #23490

### PR DESCRIPTION
**Pull request by astralia**
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

### CODE REVIEW

File: libr/anal/data.c [hunk 1]
```diff
@@ -537,9 +536,8 @@ R_API const char *r_anal_data_kind(RAnal *a, ut64 addr, const ut8 *buf, int len)
 }
 
 R_API const char *r_anal_datatype_tostring(RAnalDataType t) {
+	R_RETURN_VAL_IF_FAIL (t == R_ANAL_DATATYPE_NULL, NULL);
```
Comment (astralia) [inline libr/anal/data.c:539]: I’m not sure if this check should be done this way…

File: libr/anal/data.c [hunk 2]
```diff
@@ -537,9 +536,8 @@ R_API const char *r_anal_data_kind(RAnal *a, ut64 addr, const ut8 *buf, int len)
 }
 
 R_API const char *r_anal_datatype_tostring(RAnalDataType t) {
```
Comment (trufae) [inline libr/anal/data.c:538]:
```suggestion
R_API R_NULLABLE const char *r_anal_datatype_tostring(RAnalDataType t) {
```

File: libr/anal/data.c (hunk 1 above)
Comment (trufae) [inline libr/anal/data.c:539]:
```suggestion
```

File: libr/anal/data.c [hunk 3]
```diff
@@ -537,9 +536,8 @@ R_API const char *r_anal_data_kind(RAnal *a, ut64 addr, const ut8 *buf, int len)
...
+	R_RETURN_VAL_IF_FAIL (t == R_ANAL_DATATYPE_NULL, NULL);
 	switch (t) {
-	case R_ANAL_DATATYPE_NULL:
-		return NULL;
 	case R_ANAL_DATATYPE_ARRAY:
```
Comment (trufae) [inline libr/anal/data.c:541]:
```suggestion
	case R_ANAL_DATATYPE_NULL:
		return "null";
	case R_ANAL_DATATYPE_ARRAY:
```

File: libr/anal/data.c [hunk 4]
```diff
@@ -558,6 +556,8 @@ R_API const char *r_anal_datatype_tostring(RAnalDataType t) {
 		return "int64";
 	case R_ANAL_DATATYPE_FLOAT:
 		return "float";
+	default:
```
Comment (trufae) [inline libr/anal/data.c:559]:
```suggestion
```

File: libr/anal/data.c [hunk 5]
```diff
@@ -558,6 +556,8 @@ R_API const char *r_anal_datatype_tostring(RAnalDataType t) {
...
+		break;
```
Comment (trufae) [inline libr/anal/data.c:560]:
```suggestion
```

**Review by trufae — CHANGES_REQUESTED** [review]
asdasdfasdf

File: libr/anal/data.c [hunk 6]
```diff
@@ -536,10 +535,10 @@ R_API const char *r_anal_data_kind(RAnal *a, ut64 addr, const ut8 *buf, int len)
 	return "data";
 }
 
-R_API const char *r_anal_datatype_tostring(RAnalDataType t) {
+R_API R_NULLABLE const char *r_anal_datatype_tostring(RAnalDataType t) {
 	switch (t) {
 	case R_ANAL_DATATYPE_NULL:
-		return NULL;
+		return "null";
```
Comment (trufae) [inline libr/anal/data.c:541]:
```suggestion
		return NULL;
```

**Pull request merged**